chia/wallet/wallet_state_manager.py

```diff
@@ -1303,15 +1301,11 @@ async def add_coin_states(
                                     and inner_puzhash is not None
                                     and (await self.puzzle_store.puzzle_hash_exists(inner_puzhash))
                                 ):
-                                    for _, wallet in self.wallets.items():
-                                        if wallet.type() == WalletType.DATA_LAYER.value:
-                                            assert isinstance(wallet, DataLayerWallet)
-                                            dl_wallet = wallet
-                                            break
-                                    else:  # No DL wallet exists yet
+                                    try:
+                                        dl_wallet = self.get_dl_wallet()
+                                    except ValueError:
                                         dl_wallet = await DataLayerWallet.create_new_dl_wallet(
```

Comment:
Isn't this a call that should be made under a wallet state manager lock but is not?